File: src/stores/records.ts
import { defineStore } from 'pinia';
import { ref, computed } from 'vue';
import type { AccountRecord, RecordInput } from '@/types';
import * as api from '@/api/tauri';

export const useRecordsStore = defineStore('records', () => {
  const records = ref<AccountRecord[]>([]);
  const total = ref(0);
  const page = ref(1);
  const pageSize = ref(20);
  const loading = ref(false);

  const filters = ref({
    type: '',
    category: '',
    account: '',
    datetimeGte: '',
    datetimeLte: '',
    sort: 'datetime_desc',
  });

  const totalPages = computed(() => Math.ceil(total.value / pageSize.value));

  async function fetchRecords() {
    loading.value = true;
    try {
      const params: Record<string, unknown> = {
        page: page.value,
        pageSize: pageSize.value,
        sort: filters.value.sort || 'datetime_desc',
      };
      if (filters.value.type) params.filterType = filters.value.type;
      if (filters.value.category) params.filterCategory = filters.value.category;
      if (filters.value.account) params.filterAccount = filters.value.account;
      if (filters.value.datetimeGte) params.datetimeGte = filters.value.datetimeGte;
      if (filters.value.datetimeLte) params.datetimeLte = filters.value.datetimeLte;

      console.log('[Records] fetchRecords params:', JSON.stringify(params));
      const result = await api.getRecords(params);
      console.log('[Records] 返回数据:', result.meta?.count, '条');
      records.value = result.data;
      total.value = result.meta?.count ?? 0;
    } catch (e) {
      console.error('[Records] fetchRecords 失败:', e);
    } finally {
      loading.value = false;
    }
  }

  async function createRecord(fields: RecordInput) {
    const record = await api.createRecord(fields);
    await fetchRecords();
    return record;
  }

  async function updateRecord(id: number, fields: Partial<RecordInput>) {
    await api.updateRecord(id, fields);
    await fetchRecords();
  }

  async function deleteRecord(id: number) {
    await api.deleteRecord(id);
    await fetchRecords();
  }

  return {
    records, total, page, pageSize, loading, filters, totalPages,
    fetchRecords, createRecord, updateRecord, deleteRecord,
  };
});

File: src-tauri/src/commands/records.rs
use tauri::State;

use crate::db::Database;
use crate::db::{RecordInput, RecordUpdateInput};

#[tauri::command]
pub async fn get_records(
    state: State<'_, Database>,
    page: Option<u32>,
    page_size: Option<u32>,
    filter_type: Option<String>,
    filter_category: Option<String>,
    filter_account: Option<String>,
    datetime_gte: Option<String>,
    datetime_lte: Option<String>,
    sort: Option<String>,
) -> Result<serde_json::Value, String> {
    let (records, count) = crate::db::records::get_records(
        state.inner(),
        page.unwrap_or(1),
        page_size.unwrap_or(20),
        filter_type.as_deref(),
        filter_category.as_deref(),
        filter_account.as_deref(),
        datetime_gte.as_deref(),
        datetime_lte.as_deref(),
        sort.as_deref(),
    )?;

    Ok(serde_json::json!({
        "data": records,
        "meta": { "count": count }
    }))
}

#[tauri::command]
pub async fn get_record(
    state: State<'_, Database>,
    id: i64,
) -> Result<serde_json::Value, String> {
    let record = crate::db::records::get_record(state.inner(), id)?;
    match record {
        Some(r) => Ok(serde_json::json!({ "data": r })),
        None => Err("Record not found".to_string()),
    }
}

#[tauri::command]
pub async fn create_record(
    state: State<'_, Database>,
    fields: RecordInput,
) -> Result<serde_json::Value, String> {
    let record = crate::db::records::create_record(state.inner(), fields)?;
    Ok(serde_json::json!({ "data": record }))
}

#[tauri::command]
pub async fn update_record(
    state: State<'_, Database>,
    id: i64,
    fields: RecordUpdateInput,
) -> Result<serde_json::Value, String> {
    let record = crate::db::records::update_record(state.inner(), id, fields)?;
    Ok(serde_json::json!({ "data": record }))
}

#[tauri::command]
pub async fn delete_record(
    state: State<'_, Database>,
    id: i64,
) -> Result<(), String> {
    crate::db::records::delete_record(state.inner(), id)
}
